libbeat/common/kubernetes/metadata.go

```diff
@@ -67,6 +70,21 @@ func (g *metaGenerator) PodMetadata(pod *Pod) common.MapStr {
 	return meta
 }
 
+func (g *metaGenerator) PodMetadataWithUID(pod *Pod) common.MapStr {
```

Comment:
I think this should be a setting instead, so `pod.uid` is added if requested (both by the user or your indexer

Reply:
Hey!
Thanks for the feedback. So, just to make sure I am understand correctly, we are talking about adding the `pod.uid` from the configuration, anytime that
* The `pod_uid` Indexer is used
* `pod.uid` is specified as a field inside the `fields` element?

Reply:
I was thinking more about doing something similar to the current `include_annotations` setting, for instance. Parameters are passed here: https://github.com/elastic/beats/blob/master/libbeat/common/kubernetes/metadata.go#L24

And the processor allows configuration for them here: https://github.com/elastic/beats/blob/master/libbeat/processors/add_kubernetes_metadata/config.go#L24

The indexer would pass the right parameter (enable pod.uid) without the user asking for it.

What do you think?